File: thermostat-server/src/main/java/com/opensource/state/ThermostatRecord.java
package com.opensource.state;

import java.util.UUID;

public class ThermostatRecord {
    
    private UUID deviceId;
    
    private TemperatureState temperature;

    private long timeStamp;


    public ThermostatRecord(UUID deviceId, TemperatureState temperature, long timeStamp) {
        this.deviceId = deviceId;
        this.temperature = temperature;
        this.timeStamp = timeStamp;
    }

    public ThermostatRecord() {
    }


    public UUID getDeviceId() {
        return this.deviceId;
    }

    public void setDeviceId(UUID deviceId) {
        this.deviceId = deviceId;
    }

    public TemperatureState getTemperature() {
        return this.temperature;
    }

    public void setTemperature(TemperatureState temperature) {
        this.temperature = temperature;
    }

    public long getTimeStamp() {
        return this.timeStamp;
    }

    public void setTimeStamp(long timeStamp) {
        this.timeStamp = timeStamp;
    }

   


}

File: thermostat-ui/src/types/ThermostatRecord.ts
import TemperatureState from "./TemperatureState";

/**
 * Define record of thermostat
 *
 * @author nmajorov@redhat.com
 *
 */
class ThermostatRecord {
  private deviceId: string;

  private temperature: TemperatureState;

  private timeStamp: number;

  constructor(
    ...args: [
      deviceId: string,
      temperature: TemperatureState,
      timeStamp: number
    ]
  ) {
    this.deviceId = "";
    this.temperature = new TemperatureState();
    this.timeStamp = Date.now();
  }

  getDeviceId(): string {
    return this.deviceId;
  }

  setDeviceId(deviceId: string) {
    this.deviceId = deviceId;
  }

  getTemperature(): TemperatureState {
    return this.temperature;
  }

  setTemperature(temperature: TemperatureState): void {
    this.temperature = temperature;
  }

  getTimeStamp(): number {
    return this.timeStamp;
  }

  setTimeStamp(timeStamp: number): void {
    this.timeStamp = timeStamp;
  }
}
export default ThermostatRecord;
